### ISSUE TITLE
Problems with paper format in booklet generation

### ISSUE THREAD

**Issue by JindrazPrahy** [pdfarranger/pdfarranger#875]
When I have an A4 pdf document and I create a booklet, the result is A3. This really confused me when I printed it as I was expecting to get an A4 booklet - I have to admit, I expected to get an A4 because this is the behaviour in the printing dialogue of Adobe Acrobat.

IMO it's more intuitive to get an A4 booklet out of A4 papers so this should be changed. But this is just my opinion.
In any case, I did not find a way to easily change the paper size from A3 to A4 within pdfarranger - I had to do it with some ghostcript script. I think such a functionality should be added to pdfarranger.

**Comment by kbengs** [comment]
You can change page size in the `page format` dialog. You can't select A4 but you can set the size in mm for width or height. It will maintain aspect ratio.

> IMO it's more intuitive to get an A4 booklet out of A4 papers so this should be changed

Maybe we can leave this open for a while and see if this gets more votes.

**Comment by JindrazPrahy** [comment]
In my opionion, the `page format` dialog needs an upgrade.
1. There are currently only two options: `cancel` and `ok`. I think it would be better if there was also "apply", which applies changes and doesn't close the dialog. If you want to re-size and crop the file, you have to open the dialog twice, which is inconventient. Apply also allows you to preview the changes.
2. The current process for changing the paper size from A3 to A4 is:
   1. Open the page format dialog
   2. Know that the width of an A4 paper is square root of 2 times smaller than that of an A3 paper
   3. Know that square root of two is 1.4142...
   4. Calculate 100/1.41421 = 70.71... in your head or using a different program
   5. Write that into the width field and press OK
I think we can agree this is a needlessly complicated process - that's why I said I have to do it with a ghostcript bash script and don't consider pdfarranger an adequate tool for that in its current state (unless there is some other way that I'm missing)

So I suggest:
1. Adding an `apply` button to the dialog.
2. Making it so that the box for width supports basic math operations like in Gnu Image Manipulator. There, if you write for example 100/2 and click away, the field does the math operation for you and its input transforms into 50.
3. Adding simple buttons you can click or a drop-down menu, which would allow you to easily change the format from say A3 to A4.

**Comment by angsch** [comment]
> If you want to re-size and crop the file, you have to open the dialog twice, which is inconventient.

Can you please elaborate what exactly is not working for you? I did a rescaling and a crop in a single go and got the expected result, ie a page that was scaled and cropped. Both operations are relative, so I don't see how something could ever be off.

**Comment by JindrazPrahy** [comment]
You are correct that you can do both operations at once. My thinking was that you would want to do the operations one at a time so that you can keep track of what's going on more easily.

>Both operations are relative

This assumption would be correct if the user didn't have the option to input the new width manually. They can, so they might want to *subtract* a certain width rather than scale. Let's say your document starts at 210 mm. If you want to subtract 10 mm and add a 10% left margin, then you can have a margin length of either
(210 - 10)*0.1 mm = 20 mm
or
210 * .1 mm - 10 mm = 11 mm
depending on which operation is done first - and that isn't indicated in the interface.

Though maybe this is an edge case :)

**Comment by DegrangeM** [comment]
I agree, it would be nice to have a checkbox to preserve page dimension when creating a booklet.

**Comment by kbengs** [comment]
The page format dialog now allows you to change between common page sizes. 

For the booklet "A4 + A4 = A4" request we only have two votes so far. (in over two years 😄) Closing this issue for now. Sorry.

**Comment by florencebiree** [comment]
So sad I just look for issues today, I would vote for it also: as A4 is a very common printer format, I think having an option to atways output an A4 booklet would be very useful.
